MariaDB/server
# MDEV-33750: Rename variables which are mysql_ to mariadb_ in Debian directory

- [x] *The Jira issue number for this PR is: MDEV-33750*

## Description
This more about branding PR. As in many places when reading Debian scripts MariaDB is  mentioned as MySQL (`mysql_`). As MariaDB is something else than MySQL (besides being mostly compatible upgrade path) it should be proudly be just MariaDB named variables in `debian/` directory and get rid of other naming. There is also couple other changes under `debian/` to brand this more.

## Release Notes
This does not affect to release notes. It's more mental thing.

## How can this PR be tested?
Build branch with `debian/deb-autobake.sh` and then install packages nothing should be changed or check Salsa-CI run
https://salsa.debian.org/illuusio/mariadb-server/-/pipelines/672483
which fails badly (which are be taken care of in other PR) but one can see that it works as expected.

## Basing the PR against the correct MariaDB version

## PR quality check
